source/game/system/CourseMap.cc

```diff
@@ -233,6 +233,372 @@ CourseMap::~CourseMap() {
+                    continue;
+                }
+
+                id = findRecursiveSector(pos, 1, false, next, distanceRatio, false);
+
+                if (id != -1) {
+                    return id;
+                }
+            }
+        }
+
+        // If that fails, search next -> prev
+        for (size_t i = 0; i < checkpoint->nextCount(); ++i) {
+            MapdataCheckPoint *next = checkpoint->nextPoint(i);
+
+            for (size_t j = 0; j < next->prevCount(); ++j) {
+                MapdataCheckPoint *prev = next->prevPoint(j);
+
+                if (prev == checkpoint) {
+                    continue;
+                }
+
+                id = findRecursiveSector(pos, 1, true, prev, distanceRatio, false);
+
+                if (id != -1) {
+                    return id;
+                }
+            }
+        }
+
+        // If that fails, search next
+        return searchNextCheckpoint(pos, 0, checkpoint, distanceRatio, false, false);
+    }
+
+    return id;
+}
+
+s16 CourseMap::findSectorOutsideSector(const EGG::Vector3f &pos, MapdataCheckPoint *checkpoint,
+        f32 &distanceRatio) {
+    s16 id = -1;
+
+    // Search next -> prev
+    for (size_t i = 0; i < checkpoint->nextCount(); ++i) {
+        MapdataCheckPoint *next = checkpoint->nextPoint(i);
+
+        for (size_t j = 0; j < next->prevCount(); ++j) {
+            MapdataCheckPoint *prev = next->prevPoint(j);
+
+            if (prev == checkpoint) {
+                continue;
+            }
+
+            id = findRecursiveSector(pos, 1, true, prev, distanceRatio, false);
+
+            if (id != -1) {
+                return id;
+            }
+        }
+    }
+
+    // If that fails, search prev -> next
+    for (size_t i = 0; i < checkpoint->prevCount(); ++i) {
+        MapdataCheckPoint *prev = checkpoint->prevPoint(i);
+
+        for (size_t j = 0; j < prev->nextCount(); ++j) {
+            MapdataCheckPoint *next = prev->nextPoint(j);
+
+            if (next == checkpoint) {
+                continue;
+            }
+
+            id = findRecursiveSector(pos, 1, false, next, distanceRatio, false);
+
+            if (id != -1) {
+                return id;
+            }
+        }
+    }
+
+    // If that fails, search next
+    for (size_t i = 0; i < checkpoint->nextCount(); ++i) {
+        MapdataCheckPoint *checkpoint_ = checkpoint->nextPoint(i);
+        id = findRecursiveSector(pos, 1, false, checkpoint_, distanceRatio, false);
+
+        if (id != -1) {
+            return id;
+        }
+    }
+
+    // If that fails, search prev
+    for (size_t i = 0; i < checkpoint->prevCount(); ++i) {
+        MapdataCheckPoint *checkpoint_ = checkpoint->prevPoint(i);
+        id = findRecursiveSector(pos, 1, true, checkpoint_, distanceRatio, false);
+
+        if (id != -1) {
+            return id;
+        }
+    }
+
+    return id;
+}
+
+/// If local search fails, remove depth limit and search all "loaded" checkpoints (regiobal search)
```

Comment:
```suggestion
/// If local search fails, remove depth limit and search all "loaded" checkpoints
```
This shouldn't be a Doxygen comment if we're not explicitly labeling it with `@brief` or something similar.